### ISSUE TITLE
怎么生成高质量的光流

### ISSUE THREAD

**Issue by kankj** [hzwer/arXiv2020-RIFE#99]
我用 `liteflownet`生成的光流没有你的`100sample`里的好，具体怎么提高呢，可以请教一下吗
![flow](https://user-images.githubusercontent.com/74552110/106455764-98d07580-64c7-11eb-892c-3e1970aa3bda.png)**我的**
![flow](https://user-images.githubusercontent.com/74552110/106455799-a423a100-64c7-11eb-8fbb-7ae5d097bac1.png)**原始**

**Comment by hzwer** [comment]
你好，可以用 warped img 和 gt 的 l1 loss finetune 一下；
其实我这个有点 overfit 了，实测 finetune 1 epoch 能涨 0.2~0.3 PSNR，再往后就不涨点了

**Comment by kankj** [comment]
怎么2x图片，加在哪里，我不知道/(ㄒoㄒ)/~~

**Comment by ShaofengZou** [comment]
你好，

想请教下，finetune光流的时候，使用的损失函数有哪些呢？可否提供下相关code~

**Comment by hzwer** [comment]
> 你好，
> 
> 想请教下，finetune光流的时候，使用的损失函数有哪些呢？可否提供下相关code~

flow0 = estimate(gt, img0)
flow1 = estimate(gt, img1)
w0 = warp(img0, flow0)
w1 = warp(img1, flow1)
occ0 = (torch.abs(w0 - gt).sum(1, True) < 0.1).detach().float()
occ1 = (torch.abs(w1 - gt).sum(1, True) < 0.1).detach().float()
loss_l1 = (torch.abs(w0 - gt) * occ0).mean() + (torch.abs(w1 - gt) * occ1).mean()

主要代码是这样的，用 adam 1e-5 学习率 finetune 一个 epoch

**Comment by ShaofengZou** [comment]
> > 你好，
> > 想请教下，finetune光流的时候，使用的损失函数有哪些呢？可否提供下相关code~
> 
> flow0 = estimate(gt, img0)
> flow1 = estimate(gt, img1)
> w0 = warp(img0, flow0)
> w1 = warp(img1, flow1)
> occ0 = (torch.abs(w0 - gt).sum(1, True) < 0.1).detach().float()
> occ1 = (torch.abs(w1 - gt).sum(1, True) < 0.1).detach().float()
> loss_l1 = (torch.abs(w0 - gt) * occ0).mean() + (torch.abs(w1 - gt) * occ1).mean()
> 
> 主要代码是这样的，用 adam 1e-5 学习率 finetune 一个 epoch
> 其中 estimate 是原版 liteflownet

Get it, thank you！

**Comment by ShaofengZou** [comment]
> > 你好，
> > 想请教下，finetune光流的时候，使用的损失函数有哪些呢？可否提供下相关code~
> 
> flow0 = estimate(gt, img0)
> flow1 = estimate(gt, img1)
> w0 = warp(img0, flow0)
> w1 = warp(img1, flow1)
> occ0 = (torch.abs(w0 - gt).sum(1, True) < 0.1).detach().float()
> occ1 = (torch.abs(w1 - gt).sum(1, True) < 0.1).detach().float()
> loss_l1 = (torch.abs(w0 - gt) * occ0).mean() + (torch.abs(w1 - gt) * occ1).mean()
> 
> 主要代码是这样的，用 adam 1e-5 学习率 finetune 一个 epoch
> 其中 estimate 是原版 liteflownet

hello，想再问一下，occ0 和 occ1中为啥要用detach()，去除这个detach(）会有什么影响呢？

**Comment by hzwer** [comment]
> > > 你好，
> > > 想请教下，finetune光流的时候，使用的损失函数有哪些呢？可否提供下相关code~
> > 
> > 
> > flow0 = estimate(gt, img0)
> > flow1 = estimate(gt, img1)
> > w0 = warp(img0, flow0)
> > w1 = warp(img1, flow1)
> > occ0 = (torch.abs(w0 - gt).sum(1, True) < 0.1).detach().float()
> > occ1 = (torch.abs(w1 - gt).sum(1, True) < 0.1).detach().float()
> > loss_l1 = (torch.abs(w0 - gt) * occ0).mean() + (torch.abs(w1 - gt) * occ1).mean()
> > 主要代码是这样的，用 adam 1e-5 学习率 finetune 一个 epoch
> > 其中 estimate 是原版 liteflownet
> 
> hello，想再问一下，occ0 和 occ1中为啥要用detach()，去除这个detach(）会有什么影响呢？

可能会生成 trivial 解，比如 occ0 和 occ1 全变成 1

**Comment by ShaofengZou** [comment]
trivial 解应该指的是occ0\occ1全为0吧？这样loss_l1就是0啦。



***@***.***
 
发件人： hzwer
发送时间： 2021-03-14 16:53
收件人： hzwer/arXiv2020-RIFE
抄送： ShaofengZou; Comment
主题： Re: [hzwer/arXiv2020-RIFE] 怎么生成高质量的光流 (#99)
你好，
想请教下，finetune光流的时候，使用的损失函数有哪些呢？可否提供下相关code~
flow0 = estimate(gt, img0)
flow1 = estimate(gt, img1)
w0 = warp(img0, flow0)
w1 = warp(img1, flow1)
occ0 = (torch.abs(w0 - gt).sum(1, True) < 0.1).detach().float()
occ1 = (torch.abs(w1 - gt).sum(1, True) < 0.1).detach().float()
loss_l1 = (torch.abs(w0 - gt) * occ0).mean() + (torch.abs(w1 - gt) * occ1).mean()
主要代码是这样的，用 adam 1e-5 学习率 finetune 一个 epoch
其中 estimate 是原版 liteflownet
hello，想再问一下，occ0 和 occ1中为啥要用detach()，去除这个detach(）会有什么影响呢？
可能会生成 trivial 解，比如 occ0 和 occ1 全变成 1

**Comment by hzwer** [comment]
@ShaofengZou 你说的对
我其实试了很多方法，最后都没什么影响

**Comment by ShaofengZou** [comment]
> @ShaofengZou 你说的对
> 我其实试了很多方法，最后都没什么大影响

好嘞，谢啦!(*＾-゜)v

**Comment by hzwer** [comment]
@ShaofengZou I add a finetuned parameters for you. https://drive.google.com/file/d/1XS0yR0ZGSFxf7gSVuQ9ahj658acEi2fF/view?usp=sharing

**Comment by ShaofengZou** [comment]
Oh, thank you very much !
